src/ourthings/Queueable/Mapbox.js

```diff
@@ -0,0 +1,220 @@
+     * @param {string} json.map - The name of the map that the layer is on
+     * @param {string} json.name - The name of the layer to clear
+     */
+    clearLayer(pid, json) {
+        const options = Object.assign({
+            map: 'default',
+            name: '',
+        }, json);
+
+        this.maps[options.map].map.getSource(json.name).setData({
+            type: 'FeatureCollection',
+            features: [],
+        });
+        this.finished(pid,self.queue.DEFINE.FIN_OK);
+    }
+
+    /**
+     * Zoom to the bounds of a layer
+     * @param {int} pid
+     * @param {object} json
+     * @param {string} json.map - Name of the map the layer is within
+     * @param {string} json.name - Name of the layer to zoom in to
+     * @param {int} json.padding - Padding around the layer zoom
+     */
+    zoomToBounds(pid, json) {
+        const options = Object.assign({
+            map: 'default',
+            name: '',
+            padding: 20,
+        }, json);
+
+        let coordinates = this.maps[options.map].map.getSource(json.name)._data.features[0].geometry.coordinates;
+        if(typeof coordinates[0][0] === 'object') {
+            coordinates = coordinates[0]
+        }
+        const bounds = coordinates.reduce((bounds, coord) =>  bounds.extend(coord), new MapboxGL.LngLatBounds(coordinates[0], coordinates[0]));
+        this.maps[options.map].map.fitBounds(bounds, { padding: options.padding });
+
+        this.finished(pid,self.queue.DEFINE.FIN_OK);
+    }
+
+    /**
+     * Query and highlight feature depending on the paint features
+     * @param {int} pid
+     * @param {object} json
+     * @param {string} json.map - The map that the querying layer is on
+     * @param {string} json.name - The name of the layer to query
+     * @param {object} json.paint - The pain object for querying ans styling
+     * @param {string} json.paint.type - The styling type that will be changed for highlighting
+     * @param {array} json.paint.value - The list of paint styles for querying e.g. [["get", "feature_name"], "test_name", "#333399"] This will be appended to the current style
+     */
+    paintQueryFeatures(pid, json) {
+        const options = Object.assign({
+            map: 'default',
+            name: '',
+            paint: {},
+        }, json);
+
+        let paint = JSON.parse(JSON.stringify(this.maps[options.map].layers[options.name])).defaultStyle[options.paint.type];
+        let final = null;
+        if(typeof paint === 'object') {
+            final = paint.splice([paint.length - 1], 1);
+            for (let value of options.paint.value) {
+                for (let i = 0; i < paint.length; i += 2) {
+                    if (value === paint[i]) {
+                        paint.splice(i, 2);
+                    }
+                }
+            }
+            paint = [...paint, ...options.paint.value];
+
+            paint.push(final[0]);
+        } else {
+            paint = options.paint.value;
+        }
+
```

Comment:
Can we get add to this a similar function as the OL select control?

```
function selectFunction(e) {
			self.queue.setMemory(options.prefix+'simpleSelect', e, "Session");
			if (e.deselected.length > 0 && e.selected.length === 0)
				self.queue.execute(options.prefix+"simpleDeselect");
			if (e.selected.length > 0)
				self.queue.execute(options.prefix+"simpleSelect");
		}
```

This allows us to hang actions off the select. EG display a template of the selected feature. This may be more suitable for a 'click to select' type function but this gives us flexibility to ties these select methods together